src/link/interface.ts

```diff
@@ -0,0 +1,57 @@
+/* eslint-disable @typescript-eslint/no-unnecessary-type-constraint */
+/* eslint-disable no-use-before-define */
+import type { MultihashDigest } from '../hashes/interface'
+import type { MultibaseEncoder, MultibaseDecoder, Multibase } from '../bases/interface'
+import type { Phantom, ByteView } from '../block/interface'
+
+
+export type { MultihashDigest, MultibaseEncoder, MultibaseDecoder }
+export type Version = 0 | 1
+
+export type DAG_PB = 0x70
+export type SHA_256 = 0x12
+
+/**
+ * Represents an IPLD link to a specific data of type `T`.
+ *
+ * @template T - Logical type of the data being linked to.
+ * @template C - multicodec code corresponding to a codec linked data is encoded with
+ * @template A - multicodec code corresponding to the hashing algorithm of the CID
```

Comment:
these 3 names don't match the template names used below, doesn't that matter to the compiler? probably should make them match anyway